frontend/src/features/admin-form/settings/hooks/useSecretKeyForm.tsx

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useRef } from 'react'
+import { useForm } from 'react-hook-form'
+
+import { isKeypairValid, SECRET_KEY_REGEX } from '~utils/secretKeyValidation'
+
+export const SECRET_KEY_NAME = 'secretKey'
+export const ACK_NAME = 'ack'
+export interface SecretKeyFormInputs {
+  [SECRET_KEY_NAME]: string
+  [ACK_NAME]?: boolean
+}
+
+export interface UseSecretKeyFormProps {
+  publicKey: string
+  onSecretKeyFormSubmit: ({ secretKey, ack }: SecretKeyFormInputs) => void
+  onClose: () => void
+  hasAck: boolean
+}
+
+/**
+ * Reusable hook for secret key form logic.
+ * Handles validation of secret key and supports various secret key selection methods such as file upload.
+ */
+export const useSecretKeyForm = ({
+  publicKey,
+  onClose,
+  onSecretKeyFormSubmit,
```

Comment:
/nit generally we don't need too specific with the name. With the appropriate context, it is understood that we are doing a `onSecretKeyFormSubmit`. Thus, a short `onSubmit` would suffice, similar to `onSecretKeyFormClose` vs `onClose`

Reply:
changed to `onSubmit`!